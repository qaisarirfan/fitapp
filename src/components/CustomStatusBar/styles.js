import {Platform, StatusBar, StyleSheet} from "react-native"
import {getStatusBarHeight} from "../../utils/getStatusBarHeight"

const StatusBarHeight =
  Platform.OS === "ios" ? getStatusBarHeight(true) : StatusBar.currentHeight

// CustomStatusBar: Styling here
const styles = StyleSheet.create({
  statusBar: {
    height: StatusBarHeight,
  },
})

export default styles
