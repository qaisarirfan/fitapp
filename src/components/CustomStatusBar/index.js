import React from "react"
import {View, StatusBar} from "react-native"
import PropTypes from "prop-types"
import styles from "./styles"

// CustomStatusBar Component content
export const CustomStatusBar = ({backgroundColor, ...rest}) => {
  return (
    <View style={[styles.statusBar, {backgroundColor}]}>
      <StatusBar
        translucent={false}
        backgroundColor={backgroundColor}
        {...rest}
      />
    </View>
  )
}

// CustomStatusBar Proptypes
CustomStatusBar.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
}

// CustomStatusBar Default props
CustomStatusBar.defaultProps = {}

export default CustomStatusBar
