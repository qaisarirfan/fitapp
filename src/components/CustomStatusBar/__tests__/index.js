import React from "react"
import renderer from "react-test-renderer"
import CustomStatusBar from ".."

const createProps = () => ({
  backgroundColor: "#000",
})

describe("<CustomStatusBar />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<CustomStatusBar {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
