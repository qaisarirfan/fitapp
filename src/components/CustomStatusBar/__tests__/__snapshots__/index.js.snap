// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomStatusBar /> Does not explode 1`] = `
<View
  style={
    Array [
      Object {
        "height": 20,
      },
      Object {
        "backgroundColor": "#000",
      },
    ]
  }
/>
`;
